pkg/didcomm/common/service/service.go

```diff
@@ -122,3 +132,22 @@ func (m *DIDCommMsg) ThreadID() (string, error) {
 
 	return "", ErrThreadIDNotFound
 }
+
+// MessageHeader is header of the generic message
+type MessageHeader struct {
+	*Header
+	Purpose  string `json:"purpose"`
```

Comment:
purpose is more likely be a decorator (~purpose).

Reply:
Thought of updating it later after RFC completion.

Reply:
done